### DISCUSSION TITLE
Bi-directional edge, double arrows on an edge between nodes

### DISCUSSION THREAD

**Discussion by rahst12** [jacomyal/sigma.js#1490] [General]
We're looking into how to create a bi-directional edge between two nodes.  Essentially, two nodes connected with one edge and an arrow on both ends of the edge pointing to each node.

We have a number of cases where nodes link to each other, resulting in the graph becoming cluttered with two edges pointing between the nodes.  We're hoping if we can set an edge to be bidirectional, then the graph will look cleaner and we'll yield a performance improvement as well.

Any pointers would be appreciated.

**Comment by jacomyal** [comment]
Hi @rahst12,

Sorry for the delay. The best solution before to render bi-directional edges with sigma in my opinion, as presented in this story: https://www.sigmajs.org/storybook/?path=/story/sigma-edge-curve--parallel-edges

Also, as it makes sense to have double arrows, I just pushed a new `createEdgeDoubleArrowProgram`, that should do what you need. I have taken the time to communicate on it or to document it yet, but you should be able to use it now (it's been shipped with `3.0.0-beta.39`).

**Reply by rahst12** [reply]
Amazing, thank you, we will check this out and let you know how it works.

**Reply by djseiss** [reply]
Hey @jacomyal,

We got to working on this before beta 39 and ended up with a hacky solution. This is awesome work that will really help us out! 

We do have one issue where we have multiple edges between the same nodes and therefore using the curved edge program. I think it would be very beneficial to have this built into the curved edge program as well.

**Reply by jacomyal** [reply]
Yes, it makes sense indeed. I'll add an enhancement ticket for this, until I find the time to implement it properly (shouldn't be too long). Thanks!

**Reply by jacomyal** [reply]
I just published @sigma/edge-curve [v3.1.0](https://www.npmjs.com/package/@sigma/edge-curve/v/3.1.0) with the double-arrows. It is showcased in this new story (with the straight double arrows as well):
https://www.sigmajs.org/storybook/?path=/story/sigma-edge-curve--arrow-heads
